src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -328,6 +222,78 @@ public static boolean isTransition(final VariantContext context) {
+     *
+     * @param gb the builder where we should put our newly called alleles, cannot be null
+     * @param assignmentMethod the method to use to do the assignment, cannot be null
+     * @param genotypeLikelihoods a vector of likelihoods to use if the method requires PLs, should be log10 likelihoods, cannot be null
+     * @param allelesToUse the alleles with respect to which the likelihoods are defined
+     */
+    public static void makeGenotypeCall(final int ploidy,
+                                           final GenotypeBuilder gb,
+                                           final GenotypeAssignmentMethod assignmentMethod,
+                                           final double[] genotypeLikelihoods,
+                                           final List<Allele> allelesToUse) {
+        switch ( assignmentMethod ) {
+            case DO_NOT_ASSIGN_GENOTYPES:
+                break;
+            case SET_TO_NO_CALL:
+                gb.alleles(noCallAlleles(ploidy));
+                gb.noGQ();
+                break;
+            case USE_PLS_TO_ASSIGN:
+                if ( genotypeLikelihoods == null || !isInformative(genotypeLikelihoods) ) {
+                    gb.alleles(noCallAlleles(ploidy));
+                    gb.noGQ();
+                } else {
+                    final int maxLikelihoodIndex = MathUtils.maxElementIndex(genotypeLikelihoods);
+                    final List<Allele> maxLikelihoodAlleles = GenotypeLikelihoods.getAlleles(maxLikelihoodIndex, ploidy).stream()
+                            .map(allelesToUse::get).collect(Collectors.toList());
+                    gb.alleles(maxLikelihoodAlleles);
+                    final int numAltAlleles = allelesToUse.size() - 1;
+                    if ( numAltAlleles > 0 ) {
+                        gb.log10PError(GenotypeLikelihoods.getGQLog10FromLikelihoods(maxLikelihoodIndex, genotypeLikelihoods));
+                    }
+                }
+                break;
+            case BEST_MATCH_TO_ORIGINAL:
+                throw new UnsupportedOperationException("Genotype call using BEST_MATCH_TO_ORIGINAL can only be done" +
+                        "with a different method that takes the original call as input");
+        }
+    }
+
+    /**
+     * Add the genotype call (GT) field to GenotypeBuilder using the requested {@link GenotypeAssignmentMethod}
+     *
+     * @param originalGT the original genotype calls, cannot be null
+     * @param gb the builder where we should put our newly called alleles, cannot be null
+     * @param assignmentMethod the method to use to do the assignment, cannot be null
+     * @param genotypeLikelihoods a vector of likelihoods to use if the method requires PLs, should be log10 likelihoods, cannot be null
+     * @param allelesToUse the alleles with respect to which the likelihoods are defined
+     */
+    public static void refreshGenotypeCall(final List<Allele> originalGT,
```

Comment:
I don't like the way BEST_MATCH_TO_ORIGINAL is treated differently... we do really need it? Under what circumstance wouldn't we just take the one based on the best PL after subsetting?